pkg/skaffold/runner/build_deploy_test.go

```diff
@@ -139,7 +139,10 @@ func TestBuildTestDeploy(t *testing.T) {
 			runner := createRunner(t, test.testBench, nil, artifacts)
 			bRes, err := runner.Build(ctx, ioutil.Discard, artifacts)
 			if err == nil {
-				err = runner.DeployAndLog(ctx, ioutil.Discard, bRes)
+				err = runner.Test(ctx, ioutil.Discard, bRes)
```

Comment:
can you add a test for test failure stopping the deploy from starting?

Reply:
This test has been already added here: https://github.com/GoogleContainerTools/skaffold/blob/6f3e2b244e639b7048d12db8d21f468db4bbe00c/pkg/skaffold/runner/build_deploy_test.go#L112